src/io/csv/csv-reader.cu

```diff
@@ -0,0 +1,1232 @@
+
+	cudaDeviceSynchronize();
+
+	std::cout << "\n--------------------------------\n";
+	std::cout << "Printing Results (dependent on Unified Memory) " << std::endl;
+
+	for ( int c = 0;  c < num_col; c++)
+	{
+		gdf_column *gdf = data[c];
+
+		std::cout << std::setw(2) << c << " =>  " << std::setw(50) << gdf->col_name << "\t" << gdf->dtype << "\t" << gdf->size << "\t" << gdf->null_count << std::endl;
+	}
+	std::cout << std::endl;
+
+	long x = 0;
+	for ( int i = 0; i < num_records; i++) {
+		x = start_data_idx + i;
+
+		std::cout << "\tRec[ " << x << "] is:  ";
+
+
+		for ( int c = 0;  c < num_col; c++)
+		{
+			gdf_column *gdf = data[c];
+
+
+			switch(gdf->dtype) {
+				case gdf_dtype::GDF_INT8:
+				{
+					int8_t *gdf_out = (int8_t *)gdf->data;
+					std::cout << "\t"  << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_INT16: {
+					int16_t *gdf_out = (int16_t *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_INT32:
+				{
+					int32_t *gdf_out = (int32_t *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_INT64:
+				{
+					int64_t *gdf_out = (int64_t *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_FLOAT32:
+				{
+					float *gdf_out = (float *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_FLOAT64:
+				{
+					double *gdf_out = (double *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_DATE64:
+				{
+					int64_t *gdf_out = (int64_t *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+				break;
+				case gdf_dtype::GDF_CATEGORY:
+				{
+					int32_t *gdf_out = (int32_t *)gdf->data;
+					std::cout << "\t" << gdf_out[x];
+				}
+					break;
+				case gdf_dtype::GDF_STRING:
+					break;
+				default:
+					break;
+			}
+		}
+		std::cout << std::endl;
+	}
+
+	std::cout << "--------------------------------\n\n";
+
+}
+
+
+
+void printInfoCheck(fields_info_t *info, int num_records, const char * text, int start_idx) {
+
+	cudaDeviceSynchronize();
+
+	std::cout << "\n--------------------------------\n";
+	std::cout << "Checking (dependent on Unified Memory) - " << text << std::endl;
+
+	std::cout << "\tRec Id\tCol Id\tStart Idx\tEnd Idx" << std::endl;
+
+	for ( int x = start_idx; x < num_records; x++) {
+		std::cout << "\t" << info->rec_id[x] << "\t" << info->col_id[x] << "\t" << info->start_idx[x] << "\t" << info->end_idx[x]   << std::endl;
+	}
+
+}
+
+
+void printStartPositions(raw_csv_t * raw_csv, int num_records)
+{
+	cudaDeviceSynchronize();
+
+	std::cout << "\n--------------------------------\n";
+	std::cout << "Print Start Indexes (dependent on Unified Memory) " << std::endl;
+
+	for ( int x = 0; x < num_records; x++) {
+		std::cout << "\t" <<  raw_csv->recStart[x] << std::endl;
+	}
+
+
+
+
+}
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Lots of extra whitespace at the end of the file here

Reply:
cleaned up and pushing changes